### PR TITLE
Make sure to set is_home if we are on the front page

### PR DESCRIPTION
**Pull request by mitogh**
In order to prevent any 404 error code we need to set `is_home` to `true` if.
we are on the front page, which means the main reading settings are set
to display a page that is displaying an events page, this should turn:

- `is_home`
- `is_front_page`

To `true`.

See: https://central.tri.be/issues/90001

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;Replace with new utility function tribe_is_event_front_page();&quot; | [Query.php:164](https://github.com/moderntribe/the-events-calendar/blob/dbdbce45a047895c001f6d2a20f96df0a3e8d390/src/Tribe/Query.php#L164)
:warning: | Usage of ELSE IF is discouraged; use ELSEIF instead | [Query.php:172](https://github.com/moderntribe/the-events-calendar/blob/dbdbce45a047895c001f6d2a20f96df0a3e8d390/src/Tribe/Query.php#L172)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;Replace with new utility function tribe_is_event_front_page();&quot; | [Query.php:164](https://github.com/moderntribe/the-events-calendar/blob/c7256de04f1afa9e5d0c42e8b159c71483701024/src/Tribe/Query.php#L164)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Query.php [hunk 1]
```diff
@@ -149,16 +149,29 @@ public static function parse_query( $query ) {
 			}
 
 			// never allow 404 on month view
-			if ( $query->is_main_query() && $query->get( 'eventDisplay' ) == 'month' && ! $query->is_tax && ! $query->tribe_is_event_category ) {
+			if (
+				$query->is_main_query()
+				&& $query->get( 'eventDisplay' ) == 'month'
```
Comment (bordoni) [inline src/Tribe/Query.php]: Yoda

Comment (mitogh) [reply src/Tribe/Query.php]: Right! Let me update that one as I just updated the format not those lines.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;Replace with new utility function tribe_is_event_front_page();&quot; | [Query.php:164](https://github.com/moderntribe/the-events-calendar/blob/08c717e0852fcf0392f082dbadd972c75031441d/src/Tribe/Query.php#L164)

> via jenkins/codesniffer-phpcs

**Comment by mitogh** [comment]
Added the `code review` label again as has been updated with the new utility function. 

cc @barryhughes @bordoni

File: src/Tribe/Query.php [hunk 2]
```diff
@@ -149,16 +149,23 @@ public static function parse_query( $query ) {
...
+				&& ! $query->tribe_is_event_category
+			) {
 				$query->is_post_type_archive = true;
 				$query->queried_object       = get_post_type_object( Tribe__Events__Main::POSTTYPE );
 				$query->queried_object_id    = 0;
 			}
 
-			// check if is_event_query === true and hook filter
-			if ( $query->tribe_is_event_query ) {
+			$events_as_front_page = tribe_get_option( 'front_page_event_archive', false );
```
Comment (barryhughes) [inline src/Tribe/Query.php]: Do we need this/can we kill this line? Not sure I see any places where we reference this var.

Comment (mitogh) [reply src/Tribe/Query.php]: Woops...seems like we don't. Let me update it.

**Pull request merged**